document-generator-prosecution/src/test/java/uk/gov/companieshouse/document/generator/prosecution/handler/ProsecutionHandlerTest.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.companieshouse.document.generator.prosecution.handler;
+
+import com.sun.org.apache.regexp.internal.RE;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.companieshouse.api.model.prosecution.prosecutioncase.ProsecutionCaseStatusApi;
+import uk.gov.companieshouse.document.generator.interfaces.model.DocumentInfoResponse;
+import uk.gov.companieshouse.document.generator.prosecution.ProsecutionType;
+import uk.gov.companieshouse.document.generator.prosecution.exception.HandlerException;
+import uk.gov.companieshouse.document.generator.prosecution.exception.ProsecutionServiceException;
+import uk.gov.companieshouse.document.generator.prosecution.mapping.model.ProsecutionDocument;
+import uk.gov.companieshouse.document.generator.prosecution.mapping.model.defendant.Defendant;
+import uk.gov.companieshouse.document.generator.prosecution.mapping.model.offence.Offence;
+import uk.gov.companieshouse.document.generator.prosecution.mapping.model.prosecutioncase.ProsecutionCase;
+import uk.gov.companieshouse.document.generator.prosecution.service.ProsecutionService;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class ProsecutionHandlerTest {
```

Comment:
Missing test coverage for:

1. Correct handling of exception when getting offences
2. Correct handling of exception when getting prosecution cases
3. Correct handling of exception if a prosecution case is in an invalid status for document generation

Reply:
This has been added as part of the refactor changes that occurred for the ProsecutionCaseStatus